pkg/util/bufalloc/byte_allocator.go

```diff
@@ -17,13 +17,15 @@ package bufalloc
 // indicates the total amount of memory and len() is the amount already
 // allocated. The size of the buffer to allocate from is grown exponentially
 // when it runs out of room up to a maximum size (chunkAllocMaxSize).
-type ByteAllocator []byte
+type ByteAllocator struct {
+	B []byte
```

Comment:
does B have to be exported? It looks like it's only ever accessed from the outside to set the cap to zero, but you could introduce a method to do this (`Truncate()`).

Reply:
Agreed. We should expose new methods if necessary, instead of exporting the field.